### PR TITLE
ensure created directories are readable/executable

### PR DESCRIPTION
**Pull request by bparees**
bug 1467819

https://bugzilla.redhat.com/show_bug.cgi?id=1467819

### CODE REVIEW

**Comment by bparees** [comment]
@csrwng ptal, after recent changes to how we manage directories in tar, our own directories were no longer group/other read/executable meaning we couldn't run the /tmp/script/run script on container start with a random uid.

File: pkg/test/fs.go [hunk 1]
```diff
@@ -106,6 +106,12 @@ func (f *FakeFileSystem) MkdirAll(dirname string) error {
 	return f.MkdirAllError
 }
 
+// MkdirAllWithPermissions creates a new directories on the fake filesystem
```
Comment (coreydaley) [inline pkg/test/fs.go]: Update comment with correct function name

**Comment by bparees** [comment]
[test]

**Comment by csrwng** [comment]
LGTM

**Comment by bparees** [comment]
@csrwng added a TC for running overridden assemble/run scripts.

**Comment by csrwng** [comment]
still lgtm

**Comment by bparees** [comment]
tests won't pass until this commit merges, so just manually merging.

**Pull request merged**